src/Microsoft.Dnx.TestHost/Project.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.Dnx.TestHost
+{
+    public class Project
+    {
+        private const string ProjectFileName = "project.json";
+
+        public string Name { get; set; }
+
+        public IDictionary<string, string> Commands { get; }
+            = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+        public static bool TryGetProject(string inputPath, out Project project)
+        {
+            project = null;
+
+            string projectDirPath;
+            if (string.Equals(Path.GetFileName(inputPath), ProjectFileName, StringComparison.OrdinalIgnoreCase))
+            {
+                projectDirPath = Path.GetDirectoryName(inputPath);
+            }
+            else
+            {
+                projectDirPath = inputPath;
+            }
+
+            // Assume the directory name is the project name if none was specified
+            var projectName = GetDirectoryName(projectDirPath);
+            var projectJsonPath = Path.Combine(projectDirPath, ProjectFileName);
+            projectJsonPath = Path.GetFullPath(projectJsonPath);
+
+            if (!File.Exists(projectJsonPath))
+            {
+                return false;
+            }
+
+            project = GetProject(File.OpenRead(projectJsonPath), projectJsonPath, projectName);
+
+            return true;
+        }
+
+        // To enable unit testing
+        internal static Project GetProject(Stream stream, string projectJsonPath, string projectName)
+        {
+            Project project;
+
+            try
+            {
+                var seriliazer = new JsonSerializer();
+                project = seriliazer.Deserialize<Project>(new JsonTextReader(new StreamReader(stream))
+                {
+                    CloseInput = true
+                });
+                project.Name = projectName;
+            }
+            catch (JsonReaderException ex)
+            {
+                throw new InvalidOperationException(
+                    $"The JSON file '{projectJsonPath}' can't be deserialized to a JSON object.", innerException: ex);
+            }
+
+            return project;
+        }
+
+        private static string GetDirectoryName(string path)
+        {
+            path = path.TrimEnd(Path.DirectorySeparatorChar);
+
+            return path
+                .Substring(Path.GetDirectoryName(path).Length)
+                .Trim(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+        }
```

Comment:
If you noticed my latest comment, `Path.GetDirectoryName` gives a path and not the name as the name suggets (yes it misleading)...btw I followed what Dnx's Project api was doing internally...related links:
https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Runtime/Project.cs#L132
https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs#L168